x-pack/plugins/security_solution/public/management/cypress/tasks/response_actions.ts

```diff
@@ -139,6 +139,38 @@ export const waitForActionToComplete = (
     });
 };
 
+export const waitForActionToSucceed = (
+  actionId: string,
+  timeout = 180000
+): Cypress.Chainable<ActionDetails> => {
+  let action: ActionDetails | undefined;
+
+  return cy
+    .waitUntil(
+      () => {
+        return request<ActionDetailsApiResponse>({
+          method: 'GET',
+          url: resolvePathVariables(ACTION_DETAILS_ROUTE, { action_id: actionId || 'undefined' }),
```

Comment:
nit: why is it needed here to fall back to `'undefined'` string value?

Reply:
that's because we always need an `action_id` when requesting details. https://github.com/elastic/kibana/blob/81381506ade1b2567c03f047305f585c43e3093d/x-pack/plugins/security_solution/common/api/endpoint/actions/details_route.ts#L10

Reply:
Oh okay I think I know what you mean by that question. 

The reason is that `waitForActionToSucceed` is a chain-able function that we use after waiting on action request (which provides the action id), which may or may not be present, hence we default it to some string here. In other words even though we've typed the `acitonId` to be a `string` in the param there's no guarantee that the argument provided would actually be so.